compiler/rustc_ast_passes/src/ast_validation.rs

```diff
@@ -122,6 +122,40 @@ impl<'a> AstValidator<'a> {
         }
     }
 
+    fn check_gat_where(
+        &self,
+        before_predicates: &[WherePredicate],
+        where_clauses: (ast::TyAliasWhereClause, ast::TyAliasWhereClause),
+    ) {
+        let sess = &self.session;
+        if !before_predicates.is_empty() {
+            let mut state = State::new();
+            if !where_clauses.1.0 {
+                state.space();
+                state.word_space("where");
+            } else {
+                state.word_space(",");
+            }
+            let mut first = true;
+            for p in before_predicates.iter() {
+                if !first {
+                    state.word_space(",");
+                }
+                first = false;
+                state.print_where_predicate(p);
+            }
+            let suggestion = state.s.eof();
+            sess.struct_span_err(where_clauses.0.1, "where clause not allowed here")
```

Comment:
hard error?

Reply:
I expected a lint

Reply:
This gets changed to a lint in a later commit.